src/mpfao/3D/tdympfao3D_ts.c

```diff
@@ -271,27 +255,28 @@ PetscErrorCode TDyMPFAOIJacobian_Vertices_3DMesh(Vec Ul, Mat A, void *ctx) {
     // For k not equal to i and j, jacobian is given as:
     // d(fluxm_ij)/dP_k =   rho_ij       + (kr/mu)_{ij,upwind}         *   T_ik  *  (1+d(rho_k)/dP_k*g*z
     //
-    for (irow=0; irow<nflux_in + npitf_bc; irow++) {
+    for (PetscInt irow=0; irow<nflux_in + npitf_bc; irow++) {
```

Comment:
I recall getting an error with this on Linux, but I need to double-check to confirm it.

Reply:
This can give you a warning (or an error, depending on your compiler flags) if you use it for a variable with the same name as one in the outer scope. But otherwise it's fine and in fact recommended for loop variables that aren't needed outside their loops.

Reply:
It's a c99 feature, but even PETSc allows this (since earlier this year). I prefer declarations at innermost scope (makes it easier to read and refactor code).

Reply:
> if you use it for a variable with the same name as one in the outer scope.

Yes, that was the case